@pytest.mark.enable_socket doesn't work

The mark to `enable_socket` doesn't seem to work.  The fixture `socket_enabled` still works.

```
pytest-socket==0.3.1
pytest==3.10.0
```

Tried also on `pytest==3.6.3` (earliest version supported according to pytest-socket metadata), same result.

MCVE `test_something.py`:

```
import socket
import pytest

def f():
    socket.socket()

@pytest.mark.enable_socket
def test_1():
    f()

def test_2(socket_enabled):
    f()
```

Demo:

```
$ pytest --disable-socket -v
======================================= test session starts =======================================
platform linux -- Python 3.7.0, pytest-3.10.0, py-1.7.0, pluggy-0.8.0 -- /tmp/example/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/example, inifile:
plugins: socket-0.3.1
collected 2 items                                                                                 

test_something.py::test_1 FAILED                                                            [ 50%]
test_something.py::test_2 PASSED                                                            [100%]

============================================ FAILURES =============================================
_____________________________________________ test_1 ______________________________________________

    @pytest.mark.enable_socket
    def test_1():
>       f()

test_something.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_something.py:5: in f
    socket.socket()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (), kwargs = {}

    def guarded(*args, **kwargs):
>       raise SocketBlockedError()
E       pytest_socket.SocketBlockedError: A test tried to use socket.socket.

.venv/lib/python3.7/site-packages/pytest_socket.py:72: SocketBlockedError
=============================== 1 failed, 1 passed in 0.04 seconds ================================```

--- a/pytest_socket.py
+++ b/pytest_socket.py
@@ -38,13 +38,20 @@ def pytest_addoption(parser):
 
 @pytest.fixture(autouse=True)
 def _socket_marker(request):
+    """
+    Create an automatically-used fixture that every test function will load.
+
+    The last option to set the fixture wins priority.
+    The expected behavior is that higher granularity options should override
+    lower granularity options.
+    """
+    if request.config.getoption('--disable-socket'):
+        request.getfixturevalue('socket_disabled')
+
     if request.node.get_closest_marker('disable_socket'):
         request.getfixturevalue('socket_disabled')
     if request.node.get_closest_marker('enable_socket'):
         request.getfixturevalue('socket_enabled')
-
-    if request.config.getoption('--disable-socket'):
-        request.getfixturevalue('socket_disabled')
 
 
 @pytest.fixture

--- a/tests/test_socket.py
+++ b/tests/test_socket.py
@@ -101,7 +101,9 @@ def test_enable_socket_marker(testdir):
             socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     """)
     result = testdir.runpytest("--verbose", "--disable-socket")
-    assert_socket_blocked(result)
+    result.assert_outcomes(1, 0, 0)
+    with pytest.raises(BaseException):
+        assert_socket_blocked(result)
 
 
 def test_urllib_succeeds_by_default(testdir):
@@ -116,6 +118,8 @@ def test_urllib_succeeds_by_default(testdir):
     """)
     result = testdir.runpytest("--verbose")
     result.assert_outcomes(1, 0, 0)
+    with pytest.raises(BaseException):
+        assert_socket_blocked(result)
 
 
 def test_enabled_urllib_succeeds(testdir):
@@ -132,7 +136,9 @@ def test_enabled_urllib_succeeds(testdir):
             assert urlopen('http://httpbin.org/get').getcode() == 200
     """)
     result = testdir.runpytest("--verbose", "--disable-socket")
-    assert_socket_blocked(result)
+    result.assert_outcomes(1, 0, 0)
+    with pytest.raises(BaseException):
+        assert_socket_blocked(result)
 
 
 def test_disabled_urllib_fails(testdir):

